### PR TITLE
Drop on-disk keybase in favor of keyring

### PR DESCRIPTION
**Pull request by alessio**
This changeset aims to adapt `gaia` to the changes introduced in
cosmos/cosmos-sdk#5180

Rationale for this work is articulated in
- https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-006-secret-store-replacement.md

______

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by alessio** [comment]
This is ready for preliminary review, although it should not be merged til cosmos/cosmos-sdk#5180 is merged first

**Comment by fedekunze** [comment]
@alessio is this R4R?

**Comment by alessio** [comment]
Yes @fedekunze, it is de facto R4R. I didn't tag it yet as such 'cause I was waiting for the SDK's respective PR to be merged first.

**EDIT**: I actually tagged it, @alexanderbez rightly removed the tag because it was blocked on the SDK's PR. But please feel free to review it, it's in fact ready and should be used to test the new keyring integration.

File: lcd_test/helpers.go
Comment (fedekunze) [inline lcd_test/helpers.go:380]: why the change?

File: lcd_test/helpers_test.go
Comment (fedekunze) [inline lcd_test/helpers_test.go:265]: do we need to create a new instance here?

File: lcd_test/helpers_test.go
Comment (fedekunze) [inline lcd_test/helpers_test.go:312]: ditto

File: lcd_test/helpers_test.go
Comment (fedekunze) [inline lcd_test/helpers_test.go:372]: ditto

File: cmd/gaiad/testnet.go
Comment (fedekunze) [inline cmd/gaiad/testnet.go:165]: ditto, is it not required anymore?

**Review by fedekunze — COMMENTED** [review]
Changes LGTM, I have some questions only. Once those are replied/addressed I'll merge the implementation on the SDK

File: cmd/gaiad/testnet.go
Comment (alessio) [reply cmd/gaiad/testnet.go:165]: Nope, thanks to `COSMOS_SDK_TEST_KEYRING = y` that creates a temporary passwordless keyring

File: lcd_test/helpers.go
Comment (alessio) [reply lcd_test/helpers.go:380]: `CreateAddr()` doesn't take `password` anymore. Thus I fell back to the default password (it could be even an empty string)

File: lcd_test/helpers_test.go
Comment (alessio) [reply lcd_test/helpers_test.go:265]: This may be a change introduced inadvertently by my IDE. Will investigate shortly and eventually respond

Comment (alessio) [reply lcd_test/helpers_test.go:265]: Ok no, I've just updated the reference name cause `kb` is now variable's name taken as a function's parameter.
The temporary keybase seems to be created to generate the body of the request. In all honesty I can't tell more, I've just updated the logic to work with the new keyring

**Comment by alessio** [comment]
FYI I've carried out some code clean up.

**Review by fedekunze — APPROVED** [review]
tested ACK. @alessio mind updating this branch to latest SDK master?

**Comment by alessio** [comment]
Will update tests ASAP

**Comment by alessio** [comment]
All sorted @fedekunze

**Review by fedekunze — APPROVED** [review]
ACK. Thanks @alessio. We’ll probably have to write down the key migration guides for the next major release (cosmoshub-4) wiki and more docs for each OS.

**Pull request merged**
